handlers/life_journey.go

```diff
@@ -28,22 +30,97 @@ func (h *Handler) LifeJourney(c echo.Context) error {
 		c.Logger().Error(err)
 		return c.JSON(http.StatusBadRequest, lifeJourneyResponse)
 	}
-	ljList, err := lifejourneys.LifeJourneyService.LoadLifeJourneys()
+
+	var lifeJourneyList []models.LifeJourney
+	lifeJourneyList, err := getLifeJourneyBenefits(lifeJourneyRequest.Id)
 	if err != nil {
 		c.Logger().Error(err)
 		return c.JSON(http.StatusBadRequest, lifeJourneyResponse)
 	}
 
-	if lifeJourneyRequest.Id != "" {
+	lifeJourneyResponse.LifeJourneyList = lifeJourneyList
+	return c.JSON(http.StatusOK, lifeJourneyResponse)
+}
+
+// LifeJourneyBenefits
+// @Summary Request Info on Life Journey Related Benefits
+// @ID life-journey-benefits
+// @Accept  json
+// @Produce json
+// @Success 200 {object} []models.Benefits
+// @Failure 400 {object} renderings.BenefitServiceError
+// @Failure 404 {object} renderings.BenefitServiceError
+// @Failure 500 {object} renderings.BenefitServiceError
+// @Router /lifejourneys/:id/benefits [get]
+func (h *Handler) LifeJourneyBenefits(c echo.Context) error {
+	var lifeJourneyBenefitsResponse []models.Benefits
+	lifeJourneyBenefitsRequest := new(bindings.LifeJourneyBenefitsRequest)
+
+	// Bind the request into our request struct
+	if err := c.Bind(lifeJourneyBenefitsRequest); err != nil {
+		c.Logger().Error(err)
+		return c.JSON(http.StatusBadRequest, lifeJourneyBenefitsResponse)
+	}
+
+	if lifeJourneyBenefitsRequest.Id == "" {
+		return c.JSON(http.StatusUnprocessableEntity, lifeJourneyBenefitsResponse)
+	}
+
+	var lifeJourneyList []models.LifeJourney
+	lifeJourneyList, err := getLifeJourneyBenefits(lifeJourneyBenefitsRequest.Id)
+	if err != nil {
+		c.Logger().Error(err)
+		return c.JSON(http.StatusBadRequest, lifeJourneyBenefitsResponse)
+	}
+
+	if len(lifeJourneyList) != 1 {
+		return c.JSON(http.StatusUnprocessableEntity, lifeJourneyBenefitsResponse)
+	}
```

Comment:
We should change it so that when a call to /lifejourneys/{id} returns an object instead of an array of 1. Also log the error. I also don't really like the 422 response as I don't think it applies here, but I'm not sure what to return. Something more like a 500 would be appropriate.

Reply:
I reworked Life Journey to either return an array of all or a single life journey in another function. The response error has also been updated